openmicroscopy/openmicroscopy
# Ticket 13018 (rebased onto develop)

This is the same as gh-4189 but couldn't be directly rebased onto develop.

----

This PR fixes access to image when share is inactive or expired.

To test:
 - log in as ``user-1`` and share image with ``user-5``
 - log in as ``user-5`` check if image is visible in the viewer, copy the image URL ``http://host/webclient/img_detail/IMAGE_ID/``
 - log in back to ``user-1`` and deactivate share (uncheck enable in share edit form)
 - log in back to ``user-5`` and check if share is greyed out, you should also not being able to click on it. Go to the URL ``http://host/webclient/img_detail/IMAGE_ID/``, you should get 404
 - as it is impossible to create expired share check if image in ``share 17754`` (owned by ``user-1``) is not visible by ``user-5``

Then:
 - log is as user-3 and check if in a tree inactive or expired shares and discussions are not accessible in at all.

cc: @will-moore
